scripts/build.d/pyside6

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -e
+
+SRCSYNC=${SRCSYNC:-git}
+
+if [ "${SRCSYNC}" == "tarball" ] ; then
+
+  display -e "\${SRCSYNC} ${SRCSYNC} not supported by this package"
+  exit
+
+elif [ "${SRCSYNC}" == "git" ] ; then
+
+  pkgname=pyside-setup
+  pkgbranch=${VERSION:-dev}
+  pkgfull=$pkgname
+
+  # unpack (clone)
+  syncgit https://code.qt.io/pyside ${pkgname} ${pkgbranch} ${pkgfull}
+
+else
+
+  display -e "Invalid \${SRCSYNC} ${SRCSYNC}"
+  exit
+
+fi
+
+# build.
+
+# prepare files for building.
+if [ ! -e ${DEVENVPREFIX}/bin/python3 ]; then
+  display -e "Python3 not exist!!"
+  exit
+fi
+
+PYTHON=${DEVENVPREFIX}/bin/python3
+
+if [ -z "${LLVM_INSTALL_DIR}" ] ; then
+  # set LLVM_INSTALL_DIR to select the libclang installation for shiboken.
+  # For example, a brew installed libclang may be specified with:
+  # export LLVM_INSTALL_DIR=/opt/homebrew/opt/llvm
+  echo "\${LLVM_INSTALL_DIR} is not set, shiboken might not find libclang"
+fi
+
+if [ -z "${QTPATHS}" ] ; then
+  # A locally installed qt might be specified with:
+  # export QTPATHS=${HOME}/opt/qt/6.2.3/macos/bin/qtpaths
+  echo "\${QTPATHS} is not set, pyside might not find qt"
+fi
+
+pushd ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src/${pkgfull} > /dev/null
+
+  if [ -n "${PYSIDE_PATCH}" ] ; then
+    # Patch for older pyside6 with newer numpy.
+    patch -p1 << EOF
+diff --git a/sources/shiboken6/libshiboken/sbknumpyarrayconverter.cpp b/sources/shiboken6/libshiboken/sbknumpyarrayconverter.cpp
+index 996968fa1..cc25b349d 100644
+--- a/sources/shiboken6/libshiboken/sbknumpyarrayconverter.cpp
++++ b/sources/shiboken6/libshiboken/sbknumpyarrayconverter.cpp
+@@ -116,8 +116,13 @@ std::ostream &operator<<(std::ostream &str, PyArrayObject *o)
+             str << " NPY_ARRAY_NOTSWAPPED";
+         if ((flags & NPY_ARRAY_WRITEABLE) != 0)
+             str << " NPY_ARRAY_WRITEABLE";
++#if NPY_VERSION >= 0x00000010 // NPY_1_23_API_VERSION
++        if ((flags & NPY_ARRAY_WRITEBACKIFCOPY) != 0)
++            str << " NPY_ARRAY_WRITEBACKIFCOPY";
++#else
+         if ((flags & NPY_ARRAY_UPDATEIFCOPY) != 0)
+             str << " NPY_ARRAY_UPDATEIFCOPY";
++#endif
+     } else {
+         str << '0';
+     }
+EOF
+  fi
+
+  if [ -z "${PYSIDE_BUILD}" ] ; then
```

Comment:
Building pyside is a little bit trickier than other packages.  Two of the major dependencies llvm/libclang and qt are not included in devenv.  I place a guard here so that without explicitly setting the environment variable, the package doesn't build, and users will stop to inspect the warning messages printed above.

Building this on my m1max takes almost 6 minutes.  I think people will appreciate the guard.